aptos-move/block-executor/src/view.rs

```diff
@@ -916,7 +1042,17 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                                     .replace_values_with_identifiers(state_value.clone(), layout);
                                 let patched_state_value = match res {
                                     Ok((patched_state_value, _)) => {
-                                        state.read_set.borrow_mut().insert(state_key.clone());
+                                        state
+                                            .resource_read_set
+                                            .borrow_mut()
+                                            .insert(state_key.clone());
+                                        state.unsync_map.write(
```

Comment:
Don't we already write to `unsync_map` after this match?

Reply:
Done.